### ISSUE TITLE
Lack of flexibility and thoughts on refactorings

### ISSUE THREAD

**Issue by stevehodgkiss** [lotus/validations#36]
Lotus Validations requires ownership of the objects constructor (and the objects attributes) in order to work properly/validate things. This makes Lotus Validations less flexible than other validation libraries. I'd like to be able to do things like the following:

```ruby
class SignupForm
  include Virtus.model(strict: true)
  include Lotus::Validations

  attribute :name, String # virtus

  validates :name, presence: true # lotus validations
end

class SignupForm
  attr_accessor :name

  include Lotus::Validations

  validates :name, presence: true # lotus validations
end
```

Some refactoring that would help achieve this:

- Remove attribute tracking from the Lotus::Validations provided constructor, and only provide a generic accessor setter constructor that can be overwritten easily (like the one in Lotus::Entity [1]).
- Build the list of attributes and values to be validated lazily, rather than in the constructor.
- Extract a `validate` method from `attribute`, to support using other libraries for type coercion and object construction. Type coercion would only be supported through the attribute method.
- Move type coercion functionality to setters (run separately from validations).

So, Lotus Validations isn't only about validations, but also object construction and type coercion. Perhaps that functionality could be broken out into separate library?

[1] https://github.com/lotus/model/blob/master/lib/lotus/entity.rb#L178-182

**Comment by jodosha** [comment]
@apotonick To clarify: @stevehodgkiss in his PR have decoupled `.attribute` from `.validates`.
The first one, keeps the old semantic for compatibility reasons with Lotus::Controller, but now it's optional to use. The latter, skips the attribute definition, and only provides a validation DSL.

Would those changes serve to your purpose?

**Comment by apotonick** [comment]
@jodosha That is exactly what I need in Reform! That way, your `#validate` can use Reform's accessors! :beers:

**Comment by jodosha** [comment]
@apotonick :beers:

**Comment by jodosha** [comment]
Closed by #37 
/cc @apotonick

**Comment by apotonick** [comment]
Great, I am thinking about replacing AM::Validations per default in Reform with Lotus' implementation.
